lib/parser/message.cc

```diff
@@ -107,21 +43,23 @@ MessageFormattedText::MessageFormattedText(MessageFixedText text, ...)
   }
   va_list ap;
   va_start(ap, text);
-  if (NeedsSpecialFormatting(p)) {
-    string_ = SpecialFormatting(p, ap);
-  } else {
-    char buffer[256];
-    vsnprintf(buffer, sizeof buffer, p, ap);
-    string_ = buffer;
-  }
+  char buffer[256];
+  vsnprintf(buffer, sizeof buffer, p, ap);
```

Comment:
Why 256?  A call to vnsprintf will return the required size for the result if the first arg is null.

Reply:
Stack allocation is cheap.  I'll extend the code to handle overflow.